azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -115,6 +124,26 @@
         self.size: Optional[int] = size
         self.source_id: uuid.UUID = ensure_uuid(source_id)
 
+    @staticmethod
+    def from_different_descriptor(
```

Comment:
I think the name should signify that a file is being uploaded, and that could be along running operation. Maybe upload_from_different_descriptor?

Reply:
Chose the name based on consistency of the "from_file_descriptor" function in StreamDescriptor. I could change it or add comments on the function explaining it. thoughts?

Reply:
I'd argue that this method is not consistent with them, since they do relatively trivial operations, and this one uploads a file to the internet.

Reply:
@AsafMah changed it